### PR TITLE
metric: Add memdb size and key count metric

### PR DESCRIPTION
**Pull request by longfangsong**
Add size and keys count metric for memdb

### CODE REVIEW

File: internal/unionstore/memdb.go [hunk 1]
```diff
@@ -358,6 +361,7 @@ func (db *MemDB) setValue(x memdbNodeAddr, value []byte) {
 	}
 	x.vptr = db.vlog.appendValue(x.addr, x.vptr, value)
 	db.size = db.size - len(oldVal) + len(value)
+	metrics.MemdbSize.Set(float64(db.size))
```
Comment (cfzjywxk) [inline internal/unionstore/memdb.go:364]: Should `Add` be used for the `MemdbSize` as it's a server metric?

**Comment by ekexium** [comment]
There could be multiple `MemDB`s. I think Gauge is inappropriate.

**Comment by ekexium** [comment]
I'm wondering about the purpose of these metrics. It answers "how much data is in memdb" but cannot tell us a lot about "how much memory is used by memdb"

**Comment by ekexium** [comment]
We don't have a clear idea of the relationship between `db.size` and the memory allocated. I'm thinking could we export both of them so that we will be able to estimate the memory usage of certain kinds of jobs, e.g.`insert into select` a large table?

File: internal/unionstore/memdb.go [hunk 2]
```diff
@@ -563,6 +567,8 @@ func (db *MemDB) deleteNode(z memdbNodeAddr) {
 
 	db.count--
 	db.size -= int(z.klen)
+	metrics.MemdbCount.Dec()
```
Comment (disksing) [inline internal/unionstore/memdb.go:570]: I prefer `Set(db.count)`

File: internal/unionstore/memdb.go (hunk 1 above)
Comment (cfzjywxk) [reply internal/unionstore/memdb.go:364]: @longfangsong 
PTAL

File: internal/unionstore/memdb.go [hunk 3]
```diff
@@ -163,6 +164,8 @@ func (db *MemDB) Reset() {
 	db.vlogInvalid = false
 	db.size = 0
 	db.count = 0
+	metrics.MemdbSize.Set(0.0)
+	metrics.MemdbCount.Set(0.0)
```
Comment (MyonKeminta) [inline internal/unionstore/memdb.go:168]: Sorry I didn't understand it well. The metrics looks like global gauges. May there be many MemDB instances in a single process and they all `Set` values to the single gauge by themselves?

Comment (cfzjywxk) [reply internal/unionstore/memdb.go:168]: I've mentioned it above and discussed it with @longfangsong , it's needed to be changed I think.

**Comment by ekexium** [comment]
#520 tracks the memory allocated to the arena. That could be reused here.

**Comment by cfzjywxk** [comment]
@longfangsong 
Please address the above comments.

**Comment by disksing** [comment]
@cfzjywxk @MyonKeminta do we want to continue this?

**Comment by disksing** [comment]
closing due to inactivity. Please reopen it if need.

**Pull request closed without merging**